web/app/js/components/util/OctopusArms.jsx

```diff
@@ -1,65 +1,74 @@
+const halfBoxHeight = baseHeight / 2;
+const controlPoint = 10; // width and height of the control points for the bezier curves
+const inboundAlignment = controlPoint * 2;
 
-  let x2 = x1 + segmentWidth + 10;
-  let y2 = y1 + 10;
+const generateSvgComponents = (y1, width, height) => {
+  let segmentWidth = width / 2 - controlPoint; // width of each horizontal arrow segment
 
-  let x3 = x2 + 10;
-  let y3 = y2 + height + 10;
+  let x1 = 0;
+
+  let x2 = x1 + segmentWidth;
+  let x3 = x2 + controlPoint;
+
+  let y2 = y1 - controlPoint;
+  let y3 = y2 - height;
+  let y4 = y3 - controlPoint;
+
+  let x4 = x3 + controlPoint;
+  let x5 = x4 + segmentWidth;
 
-  let horizLine1 = `M ${x1},${y1} L ${x1+segmentWidth},${y1}`;
-  let curve1 = `C ${x1+segmentWidth+7},${y1} ${x2},${y1+3} ${x2},${y2}`;
-  let verticalLine = `L ${x2}, ${y2+height}`;
-  let curve2 = `C ${x2},${y2+height+6} ${x2+2},${y3} ${x3},${y3}`;
-  let horizLine2 = `L ${x3 + segmentWidth},${y3}`;
-  let arrow = `${horizLine1} ${curve1} ${verticalLine} ${curve2} ${horizLine2}`;
+  let start = `M ${x1},${y1}`;
+  let horizLine1 = `L ${x2},${y1}`;
+  let curve1 = `C ${x3},${y1} ${x3},${y1}`;
+  let curve1End = `${x3},${y2}`;
+  let verticalLineEnd = `L ${x3},${y3}`;
+  let curve2 = `C ${x3},${y4} ${x3},${y4}`;
+  let curve2End = `${x4},${y4}`;
+  let horizLine2 = `L ${x5},${y4}`;
+
+  let arrowPath = _.join([start, horizLine1, curve1, curve1End, verticalLineEnd, curve2, curve2End, horizLine2], " ");
 
   let arrowEndX = width;
-  let arrowEndY = direction === "up" ? y1 : y3;
+  let arrowEndY = y4;
   let arrowHead = `${arrowEndX - 4} ${arrowEndY - 4} ${arrowEndX} ${arrowEndY} ${arrowEndX - 4} ${arrowEndY + 4}`;
 
-  let circle = {
-    cx: x1,
-    cy: direction === "up" ? y3 : y1
-  };
+  let circle = { cx: x1, cy: y1 };
 
   return {
-    arrow,
-    arrowHead,
-    circle
+    arrowPath,
+    circle,
+    arrowHead
   };
 };
 
-const up = (width, svgHeight, arrowHeight, isOutbound) => {
-  let height = (svgHeight / 2) - arrowHeight;
+const up = (width, svgHeight, arrowHeight, isOutbound, isEven) => {
+  let height = arrowHeight + (isEven ? 0 : halfBoxHeight);
 
-  let x1 = 0;
-  let y1;
+  // up arrows start and the center of the middle node for outbound arms,
+  // and at the noce position for inbound arms
+  let y1 = isOutbound ? svgHeight / 2 : arrowHeight;
+  let arm = generateSvgComponents(y1, width, height, "up");
 
-  if (isOutbound) {
-    y1 = arrowHeight - 20; // I don't know where we intoduced that 20 but we need to match the other arrow
-  } else {
-    y1 = svgHeight / 2 ;
-  }
-
-  let svgPaths = svgArrow(x1, y1, width, height, "up");
+  let translate = isOutbound ? null : `translate(0, ${svgHeight / 2 + (isEven ? 0 : halfBoxHeight) + inboundAlignment})`;
 
   return (
-    <g key={`up-arrow-${height}`} id="downstream-up" fill="none" stroke="none" strokeWidth="1">
+    <g key={`up-arrow-${height}`} id="downstream-up" fill="none" strokeWidth="1">
```

Comment:
`up` and `down` return pretty similar `g` components, what do think about factoring those out?

in `up`:

```js
return g(height, arm, translate, null, "up");
```

in `down`:

```js
  return g(height, arm, translate, reflect, "down");
```

and then...
```js
const g = (height, arm, translate, reflect, direction) => {
  return (
    <g key={`${direction}-arrow-${height}`} id={`downstream-${direction}`} fill="none" strokeWidth="1">
      <path
        d={arm.arrowPath}
        stroke={arrowColor}
        transform={`${translate} ${reflect}`}
        strokeOpacity={strokeOpacity} />
      <circle
        cx={arm.circle.cx}
        cy={arm.circle.cy}
        transform={`${translate} ${reflect}`}
        fill={arrowColor}
        r="4" />
      <polyline
        points={arm.arrowHead}
        stroke={arrowColor}
        strokeLinecap="round"
        transform={`${translate} ${reflect}`} />
    </g>
  );
};
```